beacon-chain/core/blocks/exit.go

```diff
@@ -61,8 +63,15 @@ func ProcessVoluntaryExits(
 		if err := VerifyExitAndSignature(val, beaconState.Slot(), beaconState.Fork(), exit, beaconState.GenesisValidatorsRoot()); err != nil {
 			return nil, errors.Wrapf(err, "could not verify exit %d", idx)
 		}
-		beaconState, err = v.InitiateValidatorExit(ctx, beaconState, exit.Exit.ValidatorIndex)
-		if err != nil {
+		beaconState, exitEpoch, err = v.InitiateValidatorExit(ctx, beaconState, exit.Exit.ValidatorIndex, maxExitEpoch, churn)
+		if err == nil {
+			if exitEpoch > maxExitEpoch {
+				maxExitEpoch = exitEpoch
+				churn = 1
+			} else if exitEpoch == maxExitEpoch {
```

Comment:
can't we simplify this to `else {`? I don't think `exitEpoch < maxExitEpoch` is ever possible

Reply:
I agree, but I wanted to be safe of our future self, it's a pretty cheap check though and it's also descriptive.